pkg/util/goroutinemap/goroutinemap.go

```diff
@@ -43,34 +43,38 @@ const (
 	maxDurationBeforeRetry time.Duration = 2 * time.Minute
 )
 
-// GoRoutineMap defines the supported set of operations.
-type GoRoutineMap interface {
-	// Run adds operationName to the list of running operations and spawns a new
-	// go routine to execute the operation. If an operation with the same name
-	// already exists, an error is returned. Once the operation is complete, the
-	// go routine is terminated and the operationName is removed from the list
-	// of executing operations allowing a new operation to be started with the
-	// same name without error.
-	Run(operationName string, operationFunc func() error) error
+// GoroutineMap defines the supported set of operations.
+type GoroutineMap interface {
+	// Run adds operationName/operationSubName to the list of running operations
+	// and spawns a new go routine to execute operationFunc.
+	// If an operation with the same operationName and no sub name already
+	// exists, an AlreadyExists or ExponentialBackoff error is returned.
+	// If an operation with the same operationName and operationSubName already
+	// exists, an AlreadyExists or ExponentialBackoff error is returned.
+	// Once the operation is complete, the go routine is terminated and the
+	// operationName/operationSubName is removed from the list of executing
+	// operations allowing a new operation to be started with the same name
+	// without error.
+	Run(operationName, operationSubName string, operationFunc func() error) error
 
 	// Wait blocks until all operations are completed. This is typically
 	// necessary during tests - the test should wait until all operations finish
 	// and evaluate results after that.
 	Wait()
 }
 
-// NewGoRoutineMap returns a new instance of GoRoutineMap.
-func NewGoRoutineMap(exponentialBackOffOnError bool) GoRoutineMap {
+// NewGoroutineMap returns a new instance of GoroutineMap.
+func NewGoroutineMap(exponentialBackOffOnError bool) GoroutineMap {
 	g := &goRoutineMap{
-		operations:                make(map[string]operation),
+		operations:                make(map[string]map[string]operation),
 		exponentialBackOffOnError: exponentialBackOffOnError,
 	}
 	g.cond = sync.NewCond(g)
 	return g
 }
 
 type goRoutineMap struct {
-	operations                map[string]operation
+	operations                map[string]map[string]operation
 	exponentialBackOffOnError bool
 	cond                      *sync.Cond
 	sync.Mutex
```

Comment:
nit: we prefer `lock sync.Mutex` because it prevents Mutex from being public to the goRoutineMap, unless you have a need for that.